integrated_channels/canvas/client.py

```diff
@@ -295,20 +361,38 @@ def _get_course_id_from_integration_id(self, integration_id):
                 all_courses_response.status_code
             )
 
-        course_id = None
+        course_found = None
         for course in all_courses_response:
-            if course['integration_id'] == integration_id:
-                course_id = course['id']
+            if course['integration_id'] == edx_course_id:
+                course_found = course
                 break
+        return course_found
 
-        if not course_id:
+        # now let's try under all subaccounts
+        """url = "{}/api/v1/accounts/{}/courses/?search_term={}".format(
+            self.enterprise_configuration.canvas_base_url,
+            self.enterprise_configuration.canvas_account_id,
+            quote(edx_course_id),
+        )"""
+
+    def _get_course_id_from_edx_course_id(self, edx_course_id):
+        """
+        To obtain course ID we have to request all courses associated with the integrated
```

Comment:
This docstring scared me until I realized that it is not accurate. We do not get all courses and match ourselves, we call the remote API that does this. I would change this to say "To obtain the Canvas course ID, we have to use the Canvas API to search by edx_course_id."